function.dd

```diff
@@ -121,14 +121,64 @@ int foo();
 
 $(H4 $(LNAME2 pure-functions, Pure Functions))
 
-        $(P Pure functions are functions which cannot access global or static, mutable
-        state save through their arguments. This can enable optimizations based on the fact
-        that a pure function is guaranteed to mutate nothing which isn't passed to it,
+    $(P Pure functions are functions that cannot directly access global or static
+        mutable state. It guarantees that a pure function call won't modify any
```

Comment:
s/It guarantees/$(D pure) guarantees/

(there's no reference to anything in the previous sentence that can be addressed as "it")